src/state/opamSwitchAction.ml

```diff
@@ -174,18 +174,19 @@ let update_switch_state ?installed ?installed_roots ?reinstall ?pinned st =
   let installed = installed +! st.installed in
   let reinstall0 = Lazy.force st.reinstall in
   let reinstall = (reinstall +! reinstall0) %% installed in
-  let compiler_packages =
-    OpamPackage.Set.filter (OpamFormula.verifies st.switch_invariant) installed
-  in
```

Comment:
Can't we just keep this `compiler_packages` as-is with just the roots instead of the whole tree?